addons/sale/models/sale.py

```diff
@@ -638,6 +641,13 @@ def message_post(self, **kwargs):
             self.env.user.company_id.set_onboarding_step_done('sale_onboarding_sample_quotation_state')
         return super(SaleOrder, self.with_context(mail_post_autofollow=True)).message_post(**kwargs)
 
+    @api.multi
+    def send_order_confirmation_mail(self):
```

Comment:
1. I think you can make this method private

2. This method pretty much replaces the "force_quotation_send" method.
It's only used once in sale.py when context is 'send_email' and I think the line can be removed because I don't see anything using that 'send_email' context.
And only in 2 tests but the call can probably be replaced by your new method. To test.

3. Why use "force_send" here?
I don't see anywhere in the specs that the mail has to be sent immediately and this could slow down the payment process, IMO it's a bit dangerous.

4. The "is_online" context parameter is not very modular.
If this is intended to check that the sale_order is done through the e-commerce, the template should be overridden in the website_sale module, checking the "website_id" parameter of the sale_order.

Reply:
After second look with the team the best way to do this would be to override `has_to_be_paid` and `has_to_be_signed` in the website_sale module to return false if SO has a website_id.

Reply:
@awa-odoo 
1) Done.
2) context is 'send_mail' line is removed.
3) As per the discussion with the @fgi-odoo mail will be send immediately so, that's why force_send is used.
4) Done.